OpenRA.Mods.Common/Traits/World/PaletteFromGimpOrJascFile.cs

```diff
@@ -45,32 +46,25 @@ class PaletteFromGimpOrJascFileInfo : ITraitInfo
 		[Desc("Index set to be fully transparent/invisible.")]
 		public readonly int TransparentIndex = 0;
 
+		[Desc("Whether this palette is available for cursors.")]
+		public readonly bool CursorPalette = false;
+
 		public object Create(ActorInitializer init) { return new PaletteFromGimpOrJascFile(init.World, this); }
-	}
 
-	class PaletteFromGimpOrJascFile : ILoadsPalettes, IProvidesAssetBrowserPalettes
-	{
-		readonly World world;
-		readonly PaletteFromGimpOrJascFileInfo info;
+		string IProvidesCursorPaletteInfo.Palette { get { return CursorPalette ? Name : null; } }
 
-		public PaletteFromGimpOrJascFile(World world, PaletteFromGimpOrJascFileInfo info)
+		ImmutablePalette IProvidesCursorPaletteInfo.ReadPalette(IReadOnlyFileSystem fileSystem)
 		{
-			this.world = world;
-			this.info = info;
-		}
-
-		public void LoadPalettes(WorldRenderer wr)
-		{
-			var colors = new uint[Palette.Size];
-			using (var s = world.Map.Open(info.Filename))
+			using (var s = fileSystem.Open(Filename))
 			{
+				var colors = new uint[Palette.Size];
 				using (var lines = s.ReadAllLines().GetEnumerator())
 				{
 					if (lines == null)
```

Comment:
Lines can never be null, so this and the exception can be removed.

<img width="624" alt="Screenshot 2020-01-12 at 19 41 27" src="https://user-images.githubusercontent.com/167819/72224570-c6244b80-3573-11ea-973f-3b650e5acf00.png">

Reply:
Fixed